### PR TITLE
battery_level: custom format for on_click notification

### PR DESCRIPTION
**Pull request by Shir0kamii**
Hi,

I've implemented a custom format for the on_click notification of battery_level. You can also have different notifications when the computer is plugged or not.

I introduced a new placeholder, time_remaining, who was necessary to keep the same default notification as before.

I hope you like it !

### CODE REVIEW

File: py3status/modules/battery_level.py [hunk 1]
```diff
@@ -28,6 +28,12 @@
       default is None
     - format : string that formats the output. See placeholders below.
       default is "{icon}"
+    - format_notif_plugged : format of the notification received when you click
+      on the module while your computer is plugged
+      default is "In charge ({percent}%)"
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: "Charging ({percent}%)" sounds more appropriate

File: py3status/modules/battery_level.py [hunk 2]
```diff
@@ -28,6 +28,12 @@
       default is None
     - format : string that formats the output. See placeholders below.
       default is "{icon}"
+    - format_notif_plugged : format of the notification received when you click
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: format_notify_charging ?

File: py3status/modules/battery_level.py [hunk 3]
```diff
@@ -28,6 +28,12 @@
...
+      on the module while your computer is plugged
+      default is "In charge ({percent}%)"
+    - format_notif_discharging : format of the notification received when you
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: format_notify_discharging ?

File: py3status/modules/battery_level.py [hunk 4]
```diff
@@ -261,6 +290,7 @@ def _update_icon(self):
     def _update_full_text(self):
         self.full_text = self.format.format(ascii_bar=self.ascii_bar,
                                             icon=self.icon,
+                                            time_remaining=self.time_remaining,
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: alphabetical order plz

**Comment by Shir0kamii** [comment]
@ultrabug: I applied the changes you suggested

File: py3status/modules/battery_level.py [hunk 5]
```diff
@@ -28,6 +28,12 @@
       default is None
     - format : string that formats the output. See placeholders below.
       default is "{icon}"
+    - format_notify_charging : format of the notification received when you click
+      on the module while your computer is plugged
+      default is "In charge ({percent}%)"
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: "Charging ({percent}%)"

File: py3status/modules/battery_level.py [hunk 6]
```diff
@@ -74,6 +81,8 @@
 EMPTY_BLOCK_DISCHARGING = '⍀'
 FULL_BLOCK = '█'
 FORMAT = "{icon}"
+FORMAT_NOTIF_DISCHARGING = "{time_remaining}"
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: FORMAT_NOTIFY_DISCHARGING

File: py3status/modules/battery_level.py [hunk 7]
```diff
@@ -74,6 +81,8 @@
...
+FORMAT_NOTIF_PLUGGED = "In charge ({percent}%)"
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: FORMAT_NOTIFY_CHARGING

**Comment by ultrabug** [comment]
@Shir0kamii : almost ;)

**Comment by Shir0kamii** [comment]
@ultrabug: sorry, it should be good this time

**Comment by pydsigner** [comment]
Looks good to me.

File: py3status/modules/battery_level.py [hunk 8]
```diff
@@ -111,14 +122,32 @@ def battery_level(self, i3s_output_list, i3s_config):
...
+        if not self.notification:
+            return
+
+        if self.time_remaining:
+            format = self.format_notify_discharging
+        else:
+            format = self.format_notify_charging
+
+        message = format.format(ascii_bar=self.ascii_bar, icon=self.icon,
+                                time_remaining=self.time_remaining,
```
Comment (ultrabug) [inline py3status/modules/battery_level.py]: alphabetically you will order, padawan ;)

**Comment by ultrabug** [comment]
ahah, still one @Shir0kamii !

File: py3status/modules/battery_level.py (hunk 8 above)
Comment (pydsigner) [inline py3status/modules/battery_level.py]: Alphabetically order, padawan, will you <= fixed that for you ;-)

**Comment by Shir0kamii** [comment]
@ultrabug: I bet it's good this time !

**Comment by ultrabug** [comment]
Thanks <3 !

**Comment by FlorentDancy** [comment]
@ultrabug Will it be in version 2.9 ? :)

**Comment by ultrabug** [comment]
@FlorentDancy ofc it will !

**Comment by FlorentDancy** [comment]
@ultrabug The real question is: when will 2.9 be available on portage! :D

**Comment by ultrabug** [comment]
@FlorentDancy I have a few open PR and issues to manage before releasing it, maybe within a week or two

**Comment by FlorentDancy** [comment]
@ultrabug Good luck, keep us posted !

**Pull request merged**